addons/event_crm/tests/test_event_crm_flow.py

```diff
@@ -235,3 +237,47 @@ def test_order_rule_duplicate_lead(self):
         })
         self.assertEqual(len(self.test_rule_order.lead_ids), 1)
         self.assertEqual(len(test_rule_order_2.lead_ids), 0)
+
+    def test_action_execute_rule(self):
+        """ Test that the action execute rule creates leads on the bases of
+            - If event_id is specified then execute the rule for that event only
+            - If no event_id is specified then execute the rule for all the ongoing and future events
+        """
+        self.test_rule_attendee.write({
+            'lead_creation_trigger': 'done',
+            'event_registration_filter': ['|', ('email', 'ilike', 'abc@test.com'), ('phone', '=', '123-456-7890')],
+            'event_id': False,
+        })
+        attendee_data = [
+            ('abc@test.com', '111-111-1111'),
+            ('def@test.com', '123-456-7890')
+        ]
+        today = fields.Date.today()
+        new_dates = [
+            (today + timedelta(days=3), today + timedelta(days=5)),
+            (today, today + timedelta(days=2)),
+            (today - timedelta(days=5), today - timedelta(days=3))
+        ]
+        events = self.env['event.event'].create([{
+            'name': f'{prefix} Event',
+            'date_begin': today,
+            'date_end': today + timedelta(hours=1),
+            'registration_ids': [(0, 0, {
+                'name': f'{prefix} Attendee {i}',
+                'email': email,
+                'phone': phone
+            }) for i, (email, phone) in enumerate(attendee_data)]
+        } for prefix in ['Future', 'Ongoing', 'Ended']])
+        for event, (new_start, new_end) in zip(events, new_dates):
+            event.write({
+                'date_begin': new_start,
+                'date_end': new_end
+            })
+        self.assertEqual(len(self.test_rule_attendee.lead_ids), 0)
+        self.test_rule_attendee.action_execute_rule()
+        self.test_rule_attendee.lead_ids
+        self.assertEqual(len(self.test_rule_attendee.lead_ids), 4)
```

Comment:
It would be great to also test that the correct event are used, so here that the finished one is not taken into account.
Then after this, we could also test that executing a second time `action_execute_rule` does not create the same leads.